h/static/scripts/media-embedder.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+var URI = require('urijs');
+
+/**
+ * Return a YouTube embed (<iframe>) DOM element for the given video ID.
+ */
+function youTubeEmbed(id) {
+  var iframe = document.createElement('iframe');
+  iframe.src = 'https://www.youtube.com/embed/' + id;
+  return iframe;
+}
+
+function vimeoEmbed(id) {
+  var iframe = document.createElement('iframe');
+  iframe.src = 'https://player.vimeo.com/video/' + id;
+  return iframe;
+}
+
+/**
+ * A list of functions that return an "embed" DOM element (e.g. an <iframe>)
+ * for a given URL.
+ *
+ * Each function either returns `undefined` if it can't generate an embed for
+ * the URL, or a DOM element if it can.
+ */
+var embedGenerators = [
+  function iframeFromYouTubeWatchURL(url) {
+    if (!/^https:\/\/www.youtube.com\/watch(.*)$/.test(url)) {
+      return;
+    }
+
+    var id = URI(url).search(true).v;
```

Comment:
I'm not entirely sure a 61KiB external dependency is justified for something that could be done with a single regular expression: `[?&]v=([^&#]+)`